source/nodejs/adaptivecards/src/host-config.ts

```diff
@@ -230,17 +195,140 @@ export class ActionsConfig {
     }
 }
 
+export class ContainerStyleDefinition {
+    private getTextColorDefinitionOrDefault(obj: any, defaultValue: { default: string, subtle: string }) {
+        return new TextColorDefinition(obj ? obj : defaultValue);
+    }
+
+    backgroundColor?: string;
+
+    readonly foregroundColors = {
+        default: new TextColorDefinition(),
+        dark: new TextColorDefinition(),
+        light: new TextColorDefinition(),
+        accent: new TextColorDefinition(),
+        good: new TextColorDefinition(),
+        warning: new TextColorDefinition(),
+        attention: new TextColorDefinition()
+    };
+
+    parse(obj: any) {
+        if (obj) {
+            this.backgroundColor = obj["backgroundColor"];
+
+            if (obj.foregroundColors) {
+                this.foregroundColors.default = this.getTextColorDefinitionOrDefault(obj.foregroundColors["default"], { default: "#333333", subtle: "#EE333333" });
+                this.foregroundColors.dark = this.getTextColorDefinitionOrDefault(obj.foregroundColors["dark"], { default: "#000000", subtle: "#66000000" });
+                this.foregroundColors.light = this.getTextColorDefinitionOrDefault(obj.foregroundColors["light"], { default: "#FFFFFF", subtle: "#33000000" });
+                this.foregroundColors.accent = this.getTextColorDefinitionOrDefault(obj.foregroundColors["accent"], { default: "#2E89FC", subtle: "#882E89FC" });
+                this.foregroundColors.good = this.getTextColorDefinitionOrDefault(obj.foregroundColors["good"], { default: "#54A254", subtle: "#DD54A254" });
+                this.foregroundColors.warning = this.getTextColorDefinitionOrDefault(obj.foregroundColors["warning"], { default: "#E69500", subtle: "#DDE69500" });
+                this.foregroundColors.attention = this.getTextColorDefinitionOrDefault(obj.foregroundColors["attention"], { default: "#CC3300", subtle: "#DDCC3300" });
+            }
+        }        
+    }
+
+    constructor(obj?: any) {
+        this.parse(obj);
+    }
+}
+
+class NamedContainerStyleDefinition {
+    readonly isBuiltIn: boolean = false;
+    readonly name: string;
+    readonly style: ContainerStyleDefinition = new ContainerStyleDefinition();
+
+    constructor(name: string, isBuiltIn: boolean) {
+        this.name = name;
+        this.isBuiltIn = isBuiltIn;
+    }
+
+    toJSON() {
+        return {
+            name: this.name,
+            style: this.style
+        }
+    }
+}
+
 export class ContainerStyleSet {
-    readonly default: ContainerStyleDefinition = new ContainerStyleDefinition();
-    readonly emphasis: ContainerStyleDefinition = new ContainerStyleDefinition();
+    private _default = new NamedContainerStyleDefinition(Enums.ContainerStyle.Default, true);
+    private _emphasis = new NamedContainerStyleDefinition(Enums.ContainerStyle.Emphasis, true);
+    private _allStyles: Array<NamedContainerStyleDefinition> = [];
```

Comment:
Why not make this a dictionary?

Reply:
I have updated the code to use Map instead. Thanks!